### PR TITLE
Add minimum supported version information to database drivers

### PR DESCRIPTION
**Pull request by mbabker**
This adds the minimum supported version information to the database drivers and a method to retrieve that string.  The values currently used are MySQL 5.0.4 (the CMS minimum version at present) and SQL Server 2008 R2 (internal version 10.50.1600.1, per a discussion I had with Sudhi yesterday).  I've also modified the getVersion method for SQL Server to just return the version string instead of the full array the sqlsrv_server_info method returns.

### CODE REVIEW

File: libraries/joomla/database/database.php [hunk 1]
```diff
@@ -153,6 +153,12 @@ public static function test();
 	protected static $instances = array();
 
 	/**
+	 * @var    string  The minimum supported database version.
+	 * @since  12.1
+	 */
+	protected $dbMinimum;
```
Comment (chdemko) [inline libraries/joomla/database/database.php]: You can define this property static

File: libraries/joomla/database/database.php [hunk 2]
```diff
@@ -600,6 +606,18 @@ public function getLog()
...
 	/**
+	 * Get the minimum supported database version.
+	 *
+	 * @return  string  The minimum version number for the database driver.
+	 *
+	 * @since   12.1
+	 */
+	public function getMinimum()
+	{
+		return $this->dbMinimum;
```
Comment (chdemko) [inline libraries/joomla/database/database.php]: in that case, use

```
return static::$dbMinimum
```

File: libraries/joomla/database/database/mysql.php [hunk 3]
```diff
@@ -52,6 +52,12 @@ class JDatabaseMySQL extends JDatabase
 	protected $nullDate = '0000-00-00 00:00:00';
 
 	/**
+	 * @var    string  The minimum supported database version.
+	 * @since  12.1
+	 */
+	protected $dbMinimum = '5.0.4';
```
Comment (chdemko) [inline libraries/joomla/database/database/mysql.php]: static

File: libraries/joomla/database/database/sqlsrv.php [hunk 4]
```diff
@@ -50,6 +50,12 @@ class JDatabaseSQLSrv extends JDatabase
 	protected $nullDate = '1900-01-01 00:00:00';
 
 	/**
+	 * @var    string  The minimum supported database version.
+	 * @since  12.1
+	 */
+	protected $dbMinimum = '10.50.1600.1';
```
Comment (chdemko) [inline libraries/joomla/database/database/sqlsrv.php]: static

**Comment by chdemko** [comment]
We could add a method to JDatabase

```
public function isSupported()
{
    return version_compare($this->getVersion(), static::$dbMinimum) >= 0;
}
```

And define in JDatabase

```
abstract public function getVersion();
```

**Comment by mbabker** [comment]
getVersion is already abstract in JDatabase, working the other changes right now.

**Comment by mbabker** [comment]
Changes pushed up.

**Pull request merged**